lib/rubocop/cop/performance/use_zip_to_wrap_array_contents.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Performance
+      # Checks for `.map { |id| [id] }` and suggests replacing it with `.zip`.
+      #
+      # @example
+      #   # bad
+      #   [1, 2, 3].map { |id| [id] }
+      #
+      #   # good
+      #   [1, 2, 3].zip
+      #
+      # @example
+      #   # bad
+      #   [1, 2, 3].map { [_1] }
+      #
+      #   # good
+      #   [1, 2, 3].zip
+      #
+      # @example
+      #   # good (no offense)
+      #   [1, 2, 3].map { |id| id }
+      #
+      # @example
+      #   # good (no offense)
+      #   [1, 2, 3].map { |id| [id, id] }
+      class UseZipToWrapArrayContents < Base
+        extend AutoCorrector
+
+        MSG = 'Use `.zip` instead of `.map { |id| [id] }`.'
+
+        # @!method map_with_array?(node)
+        def_node_matcher :map_with_array?, <<~PATTERN
+          (block
+            (send _ :map)
+            (args (arg _id))
+            (array (lvar _id))
+          )
+        PATTERN
+
+        # Matcher for numblock (shorthand block with _1)
+        # @!method map_with_array_numblock?(node)
+        def_node_matcher :map_with_array_numblock?, <<~PATTERN
+          (numblock
+            (send _ :map)
+            1
+            (array (lvar _))
+          )
+        PATTERN
+
+        def on_block(node)
```

Comment:
Can you use `on_send` with `RESTRICT_ON_SEND = [:zip]` instead? That will have better performance.

Reply:
I got this working to replace on_block so its only on map blocks with  
RESTRICT_ON_SEND = [:map]

But I am stymied on how to handle `on_numblock` this way. Ideas? I can try to stitch the two different approaches together.

Reply:
Actually the distinct handling works quite nicely:

```
        def on_send(node)
          return unless node.method?(:map)

          parent = node.parent
          return unless parent

          register_offense(parent) if parent.block_type? && map_with_array?(parent)
        end

        def on_numblock(node)
          return unless node.method?(:map)
          return unless map_with_array_numblock?(node)

          register_offense(node)
        end
```